server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -404,13 +409,23 @@ public BytesReference cacheKey(CheckedBiConsumer<ShardSearchRequest, StreamOutpu
             if (differentiator != null) {
                 differentiator.accept(this, out);
             }
-            // copy it over since we don't want to share the thread-local bytes in #scratch
-            return out.copyBytes();
+
+            return new BytesArray(getHashedCacheKey(out.bytes()));
```

Comment:
I think we can use `MessageDigests.digest(out.bytes(), MessageDigests.sha256())` here rather than adding a new method?

Reply:
Ah, didn't see that that function already existed. Pushed a fix now!